### PR TITLE
fix: truncate generated slug to allowed character limit

### PR DESCRIPTION
**Pull request by RaunaqChawhan**
Resolve https://github.com/opencollective/opencollective/issues/6620

# Description

  truncates the generated slug to allowed character limit

# Screenshots

[fix_truncate_slug.webm](https://user-images.githubusercontent.com/32045200/232239604-05c81aad-98f3-4adc-abd5-89d0b10c53f4.webm)

### CODE REVIEW

**Review by Betree — COMMENTED** [review]
Great solution! 

I have some small feedback below about being explicit on the limit, but the feature works great.

File: lib/collective.lib.js [hunk 1]
```diff
@@ -90,7 +90,11 @@ export const suggestSlug = value => {
     strict: true,
   };
 
-  return trim(slugify(value, slugOptions), '-');
+  const truncateSlugToAllowedLimit = truncate(slugify(value, slugOptions), {
+    omission: '',
```
[the remark this answers is not in the document]
Comment (RaunaqChawhan) [reply lib/collective.lib.js:94]: yes, true! let me update my pull request by adding the length option.

[the remark this answers is not in the document]
Comment (RaunaqChawhan) [reply lib/collective.lib.js:94]: > Even though it's the default limit from lodash, we should make this explicit (i.e. not to break if they ever change it in the future)

@Betree Done.

**Comment by Betree** [comment]
Well done @RaunaqChawhan, thanks for your contribution

**Pull request merged**